frappe/installer.py

```diff
@@ -837,43 +829,60 @@ def partial_restore(sql_file_path, verbose=False):
 			fg="yellow",
 		)
 		warnings.warn(warn)
+	else:
+		click.secho("Unsupported database type", fg="red")
+		return
 
-	import_db_from_sql(source_sql=sql_file, verbose=verbose)
-
-	# Removing temporarily created file
-	if sql_file != sql_file_path:
-		os.remove(sql_file)
+	import_db_from_sql(source_sql=sql_file_path, verbose=verbose)
 
 
-def validate_database_sql(path, _raise=True):
-	"""Check if file has contents and if DefaultValue table exists
+def validate_database_sql(path: str, _raise: bool = True) -> None:
+	"""Check if file has contents and if `__Auth` table exists
 
 	Args:
 	        path (str): Path of the decompressed SQL file
 	        _raise (bool, optional): Raise exception if invalid file. Defaults to True.
 	"""
-	empty_file = False
-	missing_table = True
-
-	error_message = ""
 
-	if not os.path.getsize(path):
+	if path.endswith(".gz"):
+		executable_name = "zgrep"
```

Comment:
Is this widely available? Check frappe docker containers and typical ubuntu distros. If not better to write a stream using GZIP + grep.

Reply:
Yep, was present by default on Ubuntu, Debian, and `frappe/erpnext` image